### PR TITLE
Return a new `Memento` type from `get_memento()`

### PR DESCRIPTION
**Pull request by Mr0grog**
Rather than returning a requests.Response object from `WaybackClient.get_memento()`, we now return a completely new `Memento` type. It’s similar to requests.Response in a lot of ways, but it contains additional properties that are specific to mementos (like a timestamp) and has different values that reflect the historical archive rather than the exact response the Wayback Machine sent (e.g. `url` is the URL of the archived page, not `http://web.archive.org/web/<date>/<url>`; `headers` contains only the historic headers, not additional info added by the Wayback Machine).

There are two major reasons for this:

1. We can provide some more useful conveniences like the above noted bits.

2. We can change the underlying HTTP implementation, since we no longer expose it publicly. This is critical to getting thread safety, since Requests is not thread safe and it doesn’t look like that’s going to change soon or really ever be something they want to guarantee. Actually swapping out the underlying HTTP library is not intended to be part of this PR, though. That’ll be done separately.

    (It’ll be sad to lose some of Requests’s conveniences, but we aren’t making general purpose HTTP calls, so many of them aren’t as necessary to us — we have a lot more constraints around our usage.)

Note this drops a few features from #2 that I ultimately decided were a bit too complex (e.g. detailed media type parsing) and/or non-critical (e.g. parsing out all the timemap info from the `Links` header). We can add those in later.

**To-do:**

- [x] Code reorganization. I kinda feel like there was already too much in `_client.py` and this really pushes it well beyond.
- [x] Decide on `Memento.timestamp` vs. `Memento.datetime`.
- [x] Validate this API against scripts in web-monitoring-processing to make sure we aren’t leaving any majorly helpful features out or making anything too complicated.

### CODE REVIEW

**Comment by Mr0grog** [comment]
Marking this as ready to go! I think this is working reasonably well now and I’m relatively happy with it. Over in web-monitoring-processing, the [`test-wayback-0.3` branch](https://github.com/edgi-govdata-archiving/web-monitoring-processing/compare/main...test-wayback-0.3) makes use of it, and seems to work great.

**Comment by Mr0grog** [comment]
Rebased to merge cleanly after porting #53 over to the `main` branch.

**Comment by Mr0grog** [comment]
I think this is basically ready to go now. I took a stab at rewriting the release notes in a more final form here, combining all the changes to `get_memento()` into a slightly clearer, condensed writeup. Check the preview: https://wayback--52.org.readthedocs.build/en/52/release-history.html#in-development

After this lands, I’ll publish an alpha release `v0.3.0a1` so folks can play with it a bit before cutting a final release with such a big change.

File: docs/source/release-history.rst [hunk 1]
```diff
@@ -9,75 +9,142 @@ In Development
...
-     Don’t worry, though — complete Wayback URLs are still supported. This code still works fine:
+  Finally, you can now specify the *playback mode* of a memento using the ``mode`` parameter:
 
-     .. code-block:: python
+  .. code-block:: python
 
-        client.get_memento('http://web.archive.org/web/20180801000000id_/http://www.noaa.gov/')
+     client.get_memento('http://www.noaa.gov/',
+                        datetime=datetime.datetime(2018, 8, 1),
+                        mode=wayback.Mode.view)
```
Comment (danielballan) [inline docs/source/release-history.rst:81]: I am happy with the way this approached turned out.

File: wayback/_models.py [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+   Content hashed as a base 32 encoded SHA-1.
+
+.. py:attribute:: key
+
+   SURT-formatted URL
+
+.. py:attribute:: length
+
+   Size of captured content in bytes, such as :data:`2767`. This may be
+   innacurate. If the record is a "revisit record", indicated by MIME type
```
Comment (danielballan) [inline wayback/_models.py:34]:
```suggestion
   inaccurate. If the record is a "revisit record", indicated by MIME type
```

**Review by danielballan — APPROVED** [review]
One minor copyedit. Looks great.

**Comment by Mr0grog** [comment]
:shipit:  :shipit:  :shipit: / ⚓ AWEIGH!

**Pull request merged**